nav2_dwb_controller/dwb_core/src/publisher.cpp

```diff
@@ -158,63 +158,63 @@ DWBPublisher::publishEvaluation(std::shared_ptr<dwb_msgs::msg::LocalPlanEvaluati
 void
 DWBPublisher::publishTrajectories(const dwb_msgs::msg::LocalPlanEvaluation & results)
 {
-  if (!publish_trajectories_) {return;}
-  visualization_msgs::msg::MarkerArray ma;
-  visualization_msgs::msg::Marker m;
-
-  if (results.twists.size() == 0) {return;}
+  if (node_->count_subscribers(marker_pub_->get_topic_name()) > 0) {
```

Comment:
Sorry, not trying to be too nit-picky, but for consistency can you follow the conventions already here:

`if (!publish_trajectories_) {return;}` -> `if (node_->count_subscribers(marker_pub_->get_topic_name()) < 1) {return;}`

I really dislike having deeply nested functions for logical checks.

Reply:
Do you mind if `if (results.twists.size() == 0) {return;}` is brought out of the `if` too.

Reply:
It should all be outside the if, that's what I'm suggesting. Follow the conventions in the file of checking of a failure condition and returning then rather than wrapping the entire function with an `if(true)`